experimental/packages/opentelemetry-instrumentation-fetch/src/fetch.ts

```diff
@@ -507,8 +507,14 @@ export class FetchInstrumentation extends InstrumentationBase<FetchInstrumentati
             const body = resClone.body;
             if (body) {
               const reader = body.getReader();
-
-              const wrappedBody = withCancelPropagation(response.body, reader);
+              const isNullBodyStatus =
+                response.status === 101 ||
+                response.status === 204 ||
+                response.status === 205 ||
+                response.status === 304;
+              const wrappedBody = isNullBodyStatus
+                ? null
+                : withCancelPropagation(response.body, reader);
```

Comment:
I was able to repro [this](https://github.com/open-telemetry/opentelemetry-js/commit/65c60d8fb15d06451d5d9f92551a23a401302c56#diff-3bbcd1701ff430de523bb952f95e0bf630824bcaf964e4722efec33cea0ebcd5R508-R510) on my own branch and this looks like a reasonable more general fix.